### ISSUE TITLE
[ban-ts-comment] // @ts-expect-error should be recognized

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#1709]
**Repro**

```json
{
  "rules": {
    "@typescript-eslint/ban-ts-comment": "error"
  }
}
```

In TypeScript@next (3.9):

```ts
// @ts-expect-error
let invalid: number = "nope";
```

**Expected Result**

As of https://github.com/microsoft/TypeScript/pull/36014, `// @ts-expect-error` should be recognized and able to be banned by `@typescript-eslint/ban-ts-comment`.

**Actual Result**

It isn't yet.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`| 	`2.22.1-alpha.6`
| `@typescript-eslint/parser`| 	`2.22.0`
| `TypeScript`| 	`3.8.3`
| `ESLint`| 	`6.8.0`
| `node`| 	`v12.4.0`
| `npm` | 	`6.9.0`

**Comment by bradzacher** [comment]
see: #1706
